src/inspector/v8-debugger-agent-impl.cc

```diff
@@ -1635,19 +1635,19 @@ Response V8DebuggerAgentImpl::getTopFrameLocation(Maybe<protocol::Debugger::Loca
 
 extern "C" void V8RecordReplayGetCurrentException(v8::MaybeLocal<v8::Value>* exception);
 
-Response V8DebuggerAgentImpl::getPendingException(
-    std::unique_ptr<RemoteObject>* out_exception) {
+Response V8DebuggerAgentImpl::getPendingException(Maybe<RemoteObject>* out_exception) {
   v8::MaybeLocal<v8::Value> maybe_exception;
   V8RecordReplayGetCurrentException(&maybe_exception);
 
-  CHECK(!maybe_exception.IsEmpty());
+  if(!maybe_exception.IsEmpty()) {
+    v8::Local<v8::Context> context = m_isolate->GetCurrentContext();
+    v8::Local<v8::Value> exception = maybe_exception.ToLocalChecked();
+    std::unique_ptr<RemoteObject> obj =
+      m_session->wrapObject(context, exception, String16(), false);
 
-  v8::Local<v8::Context> context = m_isolate->GetCurrentContext();
-  v8::Local<v8::Value> exception = maybe_exception.ToLocalChecked();
-  std::unique_ptr<RemoteObject> obj =
-    m_session->wrapObject(context, exception, String16(), false);
+    *out_exception = std::move(obj);
```

Comment:
I am trying to make sense of this, since I don't have much experience with V8's approach to pointers, `Maybe` etc. I tried to re-create it, but was not able to get it to work. Maybe I'm doing something wrong?
Here is my sample: https://replit.com/@Domiii/unique-ptr#main.cpp

My main concern is that this ends up copying `obj`, but not sure why we would want to copy here? Why don't we just keep it as a `std::unique_ptr` (example of empty `unique_ptr` is also in sample code above)?

Reply:
> The type `Maybe<RemoteObject>*` is generated from the declarations in js_protocol.pdl

Oh that's kinda neat. I have not tried running the protocol-to-code generator before. How did you generate that?

Reply:
No idea! I just saw that all the stuff in `v8-debugger-agent-impl.h` is `override` so that base interface has to come from somewhere!

Reply:
It's totally possible that there's something I missed to make that assignment work, I was mostly guessing on that.

> But what made you want to use Maybe<RemoteObject>* instead of unique_ptr<RemoteObject>*? I don't see that being used anywhere?

The first example that was similar to `getPendingException` that I could find is https://github.com/replayio/chromium-v8/blob/3825061176671924b8c6fb85c2404e8edb9bf2a5/include/js_protocol.pdl#L219 which is also a Maybe with https://github.com/replayio/chromium-v8/blob/3825061176671924b8c6fb85c2404e8edb9bf2a5/src/inspector/v8-debugger-agent-impl.h#L132 so that's where that came from. That also appears to apply here: https://github.com/replayio/chromium-v8/blob/master/src/inspector/v8-debugger-agent-impl.h#L101 as another example.

My assumption was that Chromium's binding generation code uses `unique_ptr` for non-optional types and `Maybe` for optional types, but I could certainly be wrong about that.

Reply:
Sorry for overlooking this. I find Github notifications somewhat unreliable :disappointed: 

I have only ever seen `unique_ptr<RemoteObject>` and `unique_ptr<protocol::Runtime::RemoteObject>`, never `Maybe<protocol::Runtime::RemoteObject>`.

I mean, if it works its fine, but from what I tested, it won't even compile. Can you test it first, or can you consider just reverting to `unique_ptr` instead?

Reply:
(forgot in previous comment, but tagging explicitely, just to be safe... @loganfsmyth )

Reply:
Also, maybe I overlooked your last comment, @loganfsmyth. Indeed, what you said about `optional` vs non-`optional` seems to be correct. Sorry about that!

Reply:
All good, I definitely felt like I didn't get my point across but never got around to revisiting